src/client/Client.cc

```diff
@@ -2287,6 +2287,7 @@ void Client::unregister_request(MetaRequest *req)
 
 void Client::put_request(MetaRequest *request)
 {
+  ceph_assert(request->ref >= 1);
```

Comment:
Can you help me understand this part from the commit:
```
but, hitting the _exact_ crash backtrace requires the request to be a in various [x]list's.
```

Reply:
Check MetaRequest structure which has some xlist items. Those are used to track the request in various structures. Now, if the request gets prematurely destructed, the backtrace will assert in that xlist item not being in any list. That's hard to achieve with synthetic tests, therefore, this assert here is good enough to catch that code bug.

Reply:
oh so this commit is not just about the tracker in focus but it's also trying) to address a known bug

Reply:
The bug can surface itself with all kinds of backtraces - xlist asserts, etc.. the buggy ref drop is the underlying issue.